contracts/instruments/RibbonETHCoveredCall.sol

```diff
@@ -0,0 +1,185 @@
+import {IWETH} from "../interfaces/IWETH.sol";
+import {ISwap} from "../interfaces/ISwap.sol";
+import {Ownable} from "../lib/Ownable.sol";
+
+import {OptionsVaultStorageV1} from "../storage/OptionsVaultStorage.sol";
+import "hardhat/console.sol";
+
+contract RibbonETHCoveredCall is DSMath, ERC20, OptionsVaultStorageV1 {
+    using ProtocolAdapter for IProtocolAdapter;
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    enum ExchangeMechanism {Unknown, AirSwap}
+    string private constant _tokenName = "Ribbon ETH Covered Call Vault";
+    string private constant _tokenSymbol = "rETH-COVCALL";
+    string private constant _adapterName = "OPYN_GAMMA";
+    address private constant _WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    ISwap private constant _swapContract =
+        ISwap(0x4572f2554421Bd64Bef1c22c8a81840E8D496BeA);
+
+    address public constant asset = _WETH;
+    ExchangeMechanism public constant exchangeMechanism =
+        ExchangeMechanism.AirSwap;
+
+    // 1% for an instant withdrawal
+    uint256 public constant instantWithdrawalFee = 0.01 ether;
+
+    // Users can withdraw for free but have to wait for 7 days
+    uint256 public constant freeWithdrawPeriod = 7 days;
+
+    // 90% locked in options protocol, 10% of the pool reserved for withdrawals
+    uint256 public constant lockedRatio = 0.9 ether;
+
+    event ManagerChanged(address oldManager, address newManager);
+
+    event Deposited(address account, uint256 amouunt);
+
+    event WriteOptions(address manager, address options, uint256 amount);
+
+    event OptionsSaleApproved(
+        address manager,
+        address options,
+        uint256 approveAmount
+    );
+
+    constructor() ERC20(_tokenName, _tokenSymbol) {}
+
+    function initialize(address _owner, address _factory) public initializer {
+        Ownable.initialize(_owner);
+        factory = IRibbonFactory(_factory);
+        feeTo = address(this);
+    }
+
+    function setManager(address _manager) public onlyOwner {
+        require(_manager != address(0), "New manager cannot be 0x0");
+        address oldManager = manager;
+        if (oldManager != address(0)) {
+            _swapContract.revokeSigner(oldManager);
+        }
+        manager = _manager;
+        _swapContract.authorizeSigner(_manager);
+
+        emit ManagerChanged(oldManager, _manager);
+    }
+
+    function depositETH() external payable nonReentrant {
+        require(msg.value > 0, "No value passed");
+        require(asset == _WETH, "Asset is not WETH");
+
+        IWETH weth = IWETH(_WETH);
+        weth.deposit{value: msg.value}();
+        _deposit(msg.value);
+    }
+
+    function deposit(uint256 amount) external nonReentrant {
+        IERC20 assetToken = IERC20(asset);
+        assetToken.safeTransferFrom(msg.sender, address(this), amount);
+        _deposit(amount);
+    }
+
+    function _deposit(uint256 amount) private {
+        uint256 total = totalBalance().sub(amount);
```

Comment:
can you comment to explain how this works? (for auditors)

why need to get total balance minus the amount someone's trying to deposit?